## **1. Load the Cleaned Data and Basic Overview**

***
> ***In this first EDA step, we load our cleaned supply chain dataset and get a quick understanding of its structure and contents.***
***
<hr style="border: 1px solid red;">

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import os
supply_chain_path = '/content/drive/My Drive/supply_chain'
print(os.listdir(supply_chain_path))

In [2]:
import pandas as pd
df = pd.read_csv('/content/drive/My Drive/supply_chain/supply_chain_data_cleaned.csv', encoding='latin-1')

<hr style="border: 1px solid black;">

In [3]:
# Quick overview
print("Rows and columns:", df.shape)
print("\nColumn names:", df.columns.tolist())

Rows and columns: (180508, 45)

Column names: ['Type', 'Days for shipping (real)', 'Days for shipment (scheduled)', 'Benefit per order', 'Sales per customer', 'Delivery Status', 'Late_delivery_risk', 'Category Id', 'Category Name', 'Customer City', 'Customer Country', 'Customer Id', 'Customer Segment', 'Customer State', 'Customer Zipcode', 'Department Id', 'Department Name', 'Latitude', 'Longitude', 'Market', 'Order City', 'Order Country', 'Order Customer Id', 'order date (DateOrders)', 'Order Id', 'Order Item Cardprod Id', 'Order Item Discount', 'Order Item Discount Rate', 'Order Item Id', 'Order Item Product Price', 'Order Item Profit Ratio', 'Order Item Quantity', 'Sales', 'Order Item Total', 'Order Profit Per Order', 'Order Region', 'Order State', 'Order Status', 'Product Card Id', 'Product Category Id', 'Product Image', 'Product Name', 'Product Price', 'shipping date (DateOrders)', 'Shipping Mode']


<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    <strong>Data Dimensions</strong>
    <ul>
        <li><strong>Rows:</strong> 180,508</li>
        <li><strong>Columns:</strong> 45</li>
    </ul>
    <hr/>
    <strong>Column Names</strong><br/>
    We have 45 columns, including order details, shipping info, product and customer fields, and financial figures.
</div>
<hr/>
<hr style="border: 1px solid black;">


In [4]:
# Show first few rows
display(df.head())

Type  Days for shipping (real)  Days for shipment (scheduled)  \
0     DEBIT                         3                              4   
1  TRANSFER                         5                              4   
2      CASH                         4                              4   
3     DEBIT                         3                              4   
4   PAYMENT                         2                              4   

   Benefit per order  Sales per customer   Delivery Status  \
0          91.250000          314.640015  Advance shipping   
1        -249.089996          311.359985     Late delivery   
2        -247.779999          309.720001  Shipping on time   
3          22.860001          304.809998  Advance shipping   
4         134.210007          298.250000  Advance shipping   

   Late_delivery_risk  Category Id   Category Name Customer City  ...  \
0                   0           73  Sporting Goods        Caguas  ...   
1                   1           73  Sporting Goods        Caguas  ...   
2                   0           73  Sporting Goods      San Jose  ...   
3                   0           73  Sporting Goods   Los Angeles  ...   
4                   0           73  Sporting Goods        Caguas  ...   

     Order Region      Order State     Order Status Product Card Id  \
0  Southeast Asia  Java Occidental         COMPLETE            1360   
1      South Asia        Rajastán          PENDING            1360   
2      South Asia        Rajastán           CLOSED            1360   
3         Oceania       Queensland         COMPLETE            1360   
4         Oceania       Queensland  PENDING_PAYMENT            1360   

   Product Category Id                                 Product Image  \
0                   73  http://images.acmesports.sports/Smart+watch    
1                   73  http://images.acmesports.sports/Smart+watch    
2                   73  http://images.acmesports.sports/Smart+watch    
3                   73  http://images.acmesports.sports/Smart+watch    
4                   73  http://images.acmesports.sports/Smart+watch    

   Product Name  Product Price  shipping date (DateOrders)   Shipping Mode  
0  Smart watch          327.75         2018-02-03 22:56:00  Standard Class  
1  Smart watch          327.75         2018-01-18 12:27:00  Standard Class  
2  Smart watch          327.75         2018-01-17 12:06:00  Standard Class  
3  Smart watch          327.75         2018-01-16 11:45:00  Standard Class  
4  Smart watch          327.75         2018-01-15 11:24:00  Standard Class  

[5 rows x 45 columns]

<hr style="border: 1px solid black;">

In [5]:
# Show summary stats for all numeric columns
display(df.describe())

Days for shipping (real)  Days for shipment (scheduled)  \
count             180508.000000                  180508.000000   
mean                   3.497673                       2.931837   
std                    1.623743                       1.374457   
min                    0.000000                       0.000000   
25%                    2.000000                       2.000000   
50%                    3.000000                       4.000000   
75%                    5.000000                       4.000000   
max                    6.000000                       4.000000   

       Benefit per order  Sales per customer  Late_delivery_risk  \
count      180508.000000       180508.000000       180508.000000   
mean           21.971563          183.101751            0.548297   
std           104.433860          120.041578            0.497663   
min         -4274.979980            7.490000            0.000000   
25%             7.000000          104.379997            0.000000   
50%            31.520000          163.990005            1.000000   
75%            64.800003          247.399994            1.000000   
max           911.799988         1939.989990            1.000000   

         Category Id    Customer Id  Customer Zipcode  Department Id  \
count  180508.000000  180508.000000     180508.000000  180508.000000   
mean       31.849270    6690.778857      35920.246698       5.443454   
std        15.637996    4162.304984      37542.214613       1.629175   
min         2.000000       1.000000        603.000000       2.000000   
25%        18.000000    3258.000000        725.000000       4.000000   
50%        29.000000    6456.000000      19380.000000       5.000000   
75%        45.000000    9779.000000      78202.500000       7.000000   
max        76.000000   20757.000000      99205.000000      12.000000   

            Latitude  ...  Order Item Id  Order Item Product Price  \
count  180508.000000  ...  180508.000000             180508.000000   
mean       29.719858  ...   90254.756205                141.222139   
std         9.813737  ...   52108.746680                139.726960   
min       -33.937553  ...       1.000000                  9.990000   
25%        18.265432  ...   45127.750000                 50.000000   
50%        33.144863  ...   90254.500000                 59.990002   
75%        39.279617  ...  135381.250000                199.990005   
max        48.781933  ...  180519.000000               1999.989990   

       Order Item Profit Ratio  Order Item Quantity          Sales  \
count            180508.000000        180508.000000  180508.000000   
mean                  0.120639             2.127706     203.765496   
std                   0.466806             1.453469     132.270788   
min                  -2.750000             1.000000       9.990000   
25%                   0.080000             1.000000     119.980003   
50%                   0.270000             1.000000     199.919998   
75%                   0.360000             3.000000     299.950012   
max                   0.500000             5.000000    1999.989990   

       Order Item Total  Order Profit Per Order  Product Card Id  \
count     180508.000000           180508.000000    180508.000000   
mean         183.101751               21.971563       692.469414   
std          120.041578              104.433860       336.417349   
min            7.490000            -4274.979980        19.000000   
25%          104.379997                7.000000       403.000000   
50%          163.990005               31.520000       627.000000   
75%          247.399994               64.800003      1004.000000   
max         1939.989990              911.799988      1363.000000   

       Product Category Id  Product Price  
count        180508.000000  180508.000000  
mean             31.849270     141.222139  
std              15.637996     139.726960  
min               2.000000       9.990000  
25%              18.000000      50.000000  
50%        

<hr style="border: 1px solid black;">

In [6]:
# Show summary for all object/category columns
display(df.describe(include=['object', 'category']))

Type Delivery Status Category Name Customer City Customer Country  \
count   180508          180508        180508        180508           180508   
unique       4               4            50           562                2   
top      DEBIT   Late delivery        Cleats        Caguas          EE. UU.   
freq     69290           98972         24551         66768           111137   

       Customer Segment Customer State Department Name  Market     Order City  \
count            180508         180508          180508  180508         180508   
unique                3             44              11       5           3597   
top            Consumer             PR        Fan Shop   LATAM  Santo Domingo   
freq              93501          69371           66861   51594           2211   

         Order Country order date (DateOrders)     Order Region Order State  \
count           180508                  180508           180508      180508   
unique             164                   65741               23        1089   
top     Estados Unidos     2016-10-15 22:23:00  Central America  Inglaterra   
freq             24840                       5            28341        6721   

       Order Status                                      Product Image  \
count        180508                                             180508   
unique            9                                                118   
top        COMPLETE  http://images.acmesports.sports/Perfect+Fitnes...   
freq          59487                                              24515   

                            Product Name shipping date (DateOrders)  \
count                             180508                     180508   
unique                               118                      63692   
top     Perfect Fitness Perfect Rip Deck        2016-12-24 07:34:00   
freq                               24515                         10   

         Shipping Mode  
count           180508  
unique               4  
top     Standard Class  
freq            107745

<hr style="border: 1px solid red;">



## **2. Visualize the Distribution of Key Variables**
***
To get a better sense of our data, we'll plot the distribution of several important variables.
> - **Histograms** are used for numeric columns (like profit, sales, quantity) to show their spread and skew.
> - **Bar plots** (count plots) are used for categorical fields (like Delivery Status, Category Name, Shipping Mode) to see the most common values.
***
<hr style="border: 1px solid red;">